spec/feature/post_spec.rb

```diff
@@ -0,0 +1,104 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :system do
+  user = User.create(name: 'John', posts_counter: 30, photo: 'https://randomuser.me/api/portraits/men/70.jpg',
+                     bio: 'Teacher from Poland.')
+
+  subject do
+    Post.new(author_id: user.id, title: 'First Post', text: 'My first post', comments_counter: 20, likes_counters: 30)
+  end
+  before { subject.save }
+
+  describe 'index page' do
+    it "I can see the user's username." do
+      visit user_posts_path(user.id)
+      page.has_content?(user.name)
+    end
+
+    it "I can see the user's profile picture." do
+      visit user_posts_path(user.id)
+      page.has_css?('.img-fluid')
+    end
```

Comment:
- [ ] Dear @SaeqaSultani, for the user's profile picture, we need to test the presence of the image instead of the class. Please test the source image and assert the image is available.